### PR TITLE
Add Solr ZK Connection Options

### LINKED ISSUE #435: Allow option to pass ZKCLI_JVM_FLAGS to cloud-scripts zkcli.sh
When working with Zookeeper running on securePort, **cloud-scripts zkcli.sh** needs additional flags which can passed via  ZKCLI_JVM_FLAGS.  It would be great to allow this to pass thru helm chart options
e.g.
-Dzookeeper.clientCnxnSocket=org.apache.zookeeper.ClientCnxnSocketNetty

### PR DESCRIPTION
**Pull request by HoustonPutman**
Resolves #435

This is a bit broader than the Issue, but should be easier for users to setup.

Still to do:

- [x] Finish tests
- [x] Add helm options
- [x] Add documentation & changelog entry

### CODE REVIEW

**Comment by gerlowskija** [comment]
Alright, I've taken a crack at the remaining TODOs (tests, helm chart exposure, and docs).  This should be ready for review if anyone gets a chance!

File: controllers/controller_utils_test.go [hunk 1]
```diff
@@ -506,36 +507,32 @@ func filterVarsByName(envVars []corev1.EnvVar, f func(string) bool) []corev1.Env
...
 func testPodEnvVariables(expectedEnvVars map[string]string, foundEnvVars []corev1.EnvVar, additionalOffset ...int) {
-	testGenericPodEnvVariables(expectedEnvVars, foundEnvVars, "SOLR_OPTS", resolveOffset(additionalOffset))
+	testPodEnvVariablesWithGomega(Default, expectedEnvVars, foundEnvVars, resolveOffset(additionalOffset))
 }
 
 func testPodEnvVariablesWithGomega(g Gomega, expectedEnvVars map[string]string, foundEnvVars []corev1.EnvVar, additionalOffset ...int) {
-	testGenericPodEnvVariablesWithGomega(g, expectedEnvVars, foundEnvVars, "SOLR_OPTS", resolveOffset(additionalOffset))
-}
-
-func testMetricsPodEnvVariables(expectedEnvVars map[string]string, foundEnvVars []corev1.EnvVar, additionalOffset ...int) {
```
Comment (gerlowskija) [inline controllers/controller_utils_test.go:516]: These methods were all able to be merged, as the only thing differentiating them previously was the env-var that they required to be last.

**Review by HoustonPutman — COMMENTED** [review]
I can't approve this because it's my PR, but:

Absolutely beautiful, especially the new checks for EnvVar dependencies.

LGTM to merge!

**Review by gerlowskija — APPROVED** [review]

**Comment by gerlowskija** [comment]
Hah, but it lets me Approve despite having the latest commits on the branch 🤷 

Thanks for the review Houston!

Bit more manual testing but plan to merge this afternoon.

**Pull request merged**